vars/dockerExecute.groovy

```diff
@@ -84,15 +116,37 @@ private getDockerOptions(Map dockerEnvVars, Map dockerVolumeBind, def dockerOpti
         }
     }
 
-    if (dockerOptions instanceof CharSequence) {
-        options.add(dockerOptions.toString())
-    } else if (dockerOptions instanceof List) {
-        for (String option : dockerOptions) {
-            options.add "${option}"
+    if (dockerOptions) {
+        if (dockerOptions instanceof CharSequence) {
+            options.add(dockerOptions.toString())
+        } else if (dockerOptions instanceof List) {
+            for (String option : dockerOptions) {
+                options.add "${option}"
+            }
+        } else {
+            throw new IllegalArgumentException("Unexpected type for dockerOptions. Expected was either a list or a string. Actual type was: '${dockerOptions.getClass()}'")
```

Comment:
We should not throw exceptions in steps but use the `error` step, but I guess this is legacy coding.

Reply:
@CCFenner 

>We should not throw exceptions in steps but use the error step, but I guess this is legacy coding.

[meta] I do not fully agree. ```error``` makes nothing else than throwing the ```hudson.AbortException```. See [source code](https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/aa18bbd54a7fca22f3f4093702748ed8145f628e/src/main/java/org/jenkinsci/plugins/workflow/steps/ErrorStep.java#L63). Each time we throw an ```hudson.AbortException``` we can also use the error step, but on the other and it is at least not a big mistake to throw the exception directly.

The semantics of ```hudson.AbortException``` is: Hide the stack trace (from javadoc: Signals a failure where the error was anticipated and diagnosed. When this exception is caught, the stack trace will not be printed, and the build will be marked as a failure.) I can imagine cases where having the stack trace is a benefit for troubleshooting an issue. There are also other cases where we throw other exceptions than ```hudson.AbortExcpetion```, e.g. [here](https://github.com/SAP/jenkins-library/blob/2716054392ead88ecba5c984967d1ab64279db02/src/com/sap/piper/ConfigurationHelper.groovy#L15) or [here](https://github.com/SAP/jenkins-library/blob/2716054392ead88ecba5c984967d1ab64279db02/src/com/sap/piper/ConfigurationHelper.groovy#L163), and I think that is completely fine.